presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java

```diff
@@ -104,7 +105,12 @@ private DruidQueryGeneratorContext(
 
     public DruidQueryGeneratorContext withFilter(String filter)
     {
-        checkState(!hasFilter(), "Druid doesn't support filters at multiple levels");
+        if (!hasAggregation()) {
```

Comment:
To simplify the logic, could we use:
if (hasAggregation) {
    throw new PrestoException...
}
checkState(...)